lighthouse-core/scripts/i18n/collection-util.js

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env node
+/**
+* @license Copyright 2019 Google Inc. All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+*/
+'use strict';
+
+/**
+ * @typedef ICUMessageDefn
+ * @property {string} message
+ * @property {string} [description]
+ * @property {Record<string, ICUPlaceholderDefn>} [placeholders]
+ */
+
+/**
+ * @typedef ICUPlaceholderDefn
+ * @property {string} content
+ * @property {string} [example]
+ */
+
+// @ts-ignore - @types/esprima lacks all of these
+function computeDescription(ast, property, value, startRange) {
+  const endRange = property.range[0];
+  for (const comment of ast.comments || []) {
+    if (comment.range[0] < startRange) continue;
+    if (comment.range[0] > endRange) continue;
+    if (comment.value.includes('@')) {
+      // This is a complex description with description and examples.
+      let description = '';
+      /** @type {Record<string, string>} */
+      const examples = {};
+
+      const r = /@(\w+) ({\w+})?(.*)(\n|$)/g;
```

Comment:
oh is the order intentional to make the regex for `{<variableName>}` easier than example? :)

Reply:
yeah :sweat_smile: